UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -87,6 +88,24 @@ private void Update()
         movementVector.y = verticalMovement;
         characterController.Move(movementVector * Time.deltaTime);
 
+        if (characterController.isGrounded) {
+            //Check if underneath the character is stable
+            isUnderneathStable = false; //By default, underneath is not stable
+            RaycastHit hit;
+            Ray ray = new Ray(transform.position + characterController.center, Vector3.down);
+            if (Physics.Raycast(ray, out hit, Mathf.Infinity)) {
+                float underneathSlopeAngle = Vector3.Angle(transform.up, hit.normal);
+                if (underneathSlopeAngle <= characterController.slopeLimit) {
+                    //We found a stable underneath
+                    isUnderneathStable = true;
+                } else {
+                    //Add a slide movement along the underneath slope direction
+                    Vector3 underneathSlopeDirection = Vector3.Cross(hit.normal, Vector3.Cross(hit.normal, Vector3.up));
```

Comment:
Why two `Cross` operations here? Can't be done with just one?

Reply:
Doing one cross product in the current formula will only give "side" direction of the slope, but we want the "down" direction.
Maybe it could be achieved with only one cross product but I can't think about one. Do you have the formula in mind?